### DISCUSSION TITLE
variants

### DISCUSSION THREAD

**Discussion by ghost** [mtex-toolbox/mtex#506] [Ask Anything]
Hello!

Please, why the KS_T.matrix did not result in the correct 24 orientation transformation matrix?

cs_fcc = crystalSymmetry('432', [3.6599 3.6599 3.6599], 'mineral', 'Iron fcc')

cs_bcc = crystalSymmetry('432', [2.866 2.866 2.866], 'mineral', 'Iron bcc')

KS = orientation.KurdjumovSachs(cs_fcc,cs_bcc)

KS_T = KS*crystalSymmetry('432')

KS_T.matrix

**Comment by ralfHielscher** [comment]
Hi Ariane,

Could you please be a little bit more specific about what you think is
wrong and more importantly what you want to achieve. Maybe it is worth to
have a look at https://mtex-toolbox.github.io/ParentChildVariants.html and
https://mtex-toolbox.github.io/MartensiteVariants.html

Ralf.






********************************************************************
Ralf Hielscher                   Tel: +371-531-38556
Fakultät für Mathematik               +371-531-22200 (Sekr.)
Technische Universität Chemnitz  Fax: +371-531-22109
Reichenhainer Str. 39            E-mail:
ralf.hielscher@mathematik.tu-chemnitz.de
D-09126 Chemnitz                 http://www.tu-chemnitz.de/~rahi
********************************************************************

**Comment by ghost** [comment]
Hi Ralf, 
I recently started using mtex. I don't know what I missed?

**Comment by ralfHielscher** [comment]
Hi Ariane,

I was not able to understand your problem. Please write in more detail what
you want to compute.

Ralf.



********************************************************************
Ralf Hielscher                   Tel: +371-531-38556
Fakultät für Mathematik               +371-531-22200 (Sekr.)
Technische Universität Chemnitz  Fax: +371-531-22109
Reichenhainer Str. 39            E-mail:
ralf.hielscher@mathematik.tu-chemnitz.de
D-09126 Chemnitz                 http://www.tu-chemnitz.de/~rahi
********************************************************************

**Comment by ghost** [comment]
Ralf,
Thank you for looking into this. The orientation tranformation matrices of KS-OR are  KS = bcc_child * inv(fcc_parent). The 24 different KS orientation transformation matrices that correspond to the crystallographic variants are: 

![image](https://user-images.githubusercontent.com/68469500/87933246-4962b480-ca63-11ea-90f3-77f26114d0dd.png)
Fonte: https://doi.org/10.1016/j.actamat.2005.11.001

The matrices don't exactly match the answer:

cs_fcc = crystalSymmetry('432', [3.6599 3.6599 3.6599], 'mineral', 'Iron fcc');

cs_bcc = crystalSymmetry('432', [2.866 2.866 2.866], 'mineral', 'Iron bcc');

KS = orientation.KurdjumovSachs(cs_fcc,cs_bcc);

KS_T = KS*crystalSymmetry('432');

> KS_T.matrix


ans(:,:,1) =

    0.7416   -0.6667   -0.0749
    0.6498    0.7416   -0.1667
    0.1667    0.0749    0.9832


ans(:,:,2) =

   -0.6667   -0.0749    0.7416
    0.7416   -0.1667    0.6498
    0.0749    0.9832    0.1667


ans(:,:,3) =

   -0.0749    0.7416   -0.6667
   -0.1667    0.6498    0.7416
    0.9832    0.1667    0.0749


ans(:,:,4) =

   -0.6667    0.7416    0.0749
    0.7416    0.6498    0.1667
    0.0749    0.1667   -0.9832


ans(:,:,5) =

   -0.0749   -0.6667   -0.7416
   -0.1667    0.7416   -0.6498
    0.9832    0.0749   -0.1667


ans(:,:,6) =

    0.7416   -0.0749    0.6667
    0.6498   -0.1667   -0.7416
    0.1667    0.9832   -0.0749


ans(:,:,7) =

   -0.6667   -0.7416   -0.0749
    0.7416   -0.6498   -0.1667
    0.0749   -0.1667    0.9832


ans(:,:,8) =

   -0.0749    0.6667    0.7416
   -0.1667   -0.7416    0.6498
    0.9832   -0.0749    0.1667


ans(:,:,9) =

    0.7416    0.0749   -0.6667
    0.6498    0.1667    0.7416
    0.1667   -0.9832    0.0749


ans(:,:,10) =

    0.7416    0.6667    0.0749
    0.6498   -0.7416    0.1667
    0.1667   -0.0749   -0.9832


ans(:,:,11) =

   -0.6667    0.0749   -0.7416
    0.7416    0.1667   -0.6498
    0.0749   -0.9832   -0.1667


ans(:,:,12) =

   -0.0749   -0.7416    0.6667
   -0.1667   -0.6498   -0.7416
    0.9832   -0.1667   -0.0749


ans(:,:,13) =

   -0.7416    0.6667   -0.0749
   -0.6498   -0.7416   -0.1667
   -0.1667   -0.0749    0.9832


ans(:,:,14) =

    0.6667    0.0749    0.7416
   -0.7416    0.1667    0.6498
   -0.0749   -0.9832    0.1667


ans(:,:,15) =

    0.0749   -0.7416   -0.6667
    0.1667   -0.6498    0.7416
   -0.9832   -0.1667    0.0749


ans(:,:,16) =

    0.6667   -0.7416    0.0749
   -0.7416   -0.6498    0.1667
   -0.0749   -0.1667   -0.9832


ans(:,:,17) =

    0.0749    0.6667   -0.7416
    0.1667   -0.7416   -0.6498
   -0.9832   -0.0749   -0.1667


ans(:,:,18) =

   -0.7416    0.0749    0.6667
   -0.6498    0.1667   -0.7416
   -0.1667   -0.9832   -0.0749


ans(:,:,19) =

    0.6667    0.7416   -0.0749
   -0.7416    0.6498   -0.1667
   -0.0749    0.1667    0.9832


ans(:,:,20) =

    0.0749   -0.6667    0.7416
    0.1667    0.7416    0.6498
   -0.9832    0.0749    0.1667


ans(:,:,21) =

   -0.7416   -0.0749   -0.6667
   -0.6498   -0.1667    0.7416
   -0.1667    0.9832    0.0749


ans(:,:,22) =

   -0.7416   -0.6667    0.0749
   -0.6498    0.7416    0.1667
   -0.1667    0.0749   -0.9832


ans(:,:,23) =

    0.6667   -0.0749   -0.7416
   -0.7416   -0.1667   -0.6498
   -0.0749    0.9832   -0.1667


ans(:,:,24) =

    0.0749    0.7416    0.6667
    0.1667    0.6498   -0.7416
   -0.9832    0.1667   -0.0749

I recently started using mtex. So, I did some tests to understand how MTEX works.

Thank you

**Comment by ralfHielscher** [comment]
Hi Ariane,

are you referring to the fact that in the paper the numbers are rounded to 3 digits and in the MTEX output they are rounded to 4 digits? Or is there something else different? The order?

Ralf.

**Comment by ghost** [comment]
Ralf, 
Does the order impact the calculation in determining the variant?

The ans(:,:,1) = T1

ans(:,:,2) = T3

But, The ans(:,:,10) did not match with one matrix of the paper, for example.

Ariane

**Comment by ralfHielscher** [comment]
In general yes. It depends how you determine the variant. Usually, it is recommended to characterize the variants by the alignment of the axes. 

Also it is important to understand that to each of the 24 OR variants there are 24 symmetrically equivalent. This may explain the misfitting between your list and the MTEX list.

Ralf.
